circus-train-comparator/src/main/java/com/hotels/bdp/circustrain/comparator/hive/HiveDifferences.java

```diff
@@ -162,14 +171,16 @@ private HiveDifferences(
       Iterator<Partition> sourcePartitionIterator,
       Optional<Table> replicaTable,
       Optional<? extends PartitionFetcher> replicaPartitionFetcher,
-      Function<Path, String> checksumFunction) {
+      Function<Path, String> checksumFunction,
+      int partitionLimit) {
     this.diffListener = diffListener;
     this.comparatorRegistry = comparatorRegistry;
     this.sourceTable = sourceTable;
     this.sourcePartitionIterator = sourcePartitionIterator;
     this.replicaTable = replicaTable;
     this.replicaPartitionFetcher = replicaPartitionFetcher;
     this.checksumFunction = checksumFunction;
+    this.partitionLimit = partitionLimit < 0 ? UNLIMITED : partitionLimit;
```

Comment:
What if the `partitionLimit` is 0?

Reply:
then it's 0, and stays 0 and you get 0 partitions back just like you asked :)